File: app/src/androidTest/java/me/varlez/movies/common/di/TestComponent.java
package me.varlez.movies.common.di;

import javax.inject.Singleton;

import dagger.Component;
import me.varlez.movies.common.di.component.MoviesComponent;
import me.varlez.movies.movie.detail.MovieDetailsActivityBehaviorTest;
import me.varlez.movies.movie.list.MovieListActivityBehaviorTest;

/**
 * Dagger component for test purposes, using mock modules.
 */
@Singleton
@Component(modules=MockMoviesModule.class)
public interface TestComponent extends MoviesComponent {
    void inject(MovieListActivityBehaviorTest movieListActivityBehaviorTest);

    void inject(MovieDetailsActivityBehaviorTest movieDetailsActivityBehaviorTest);
}

File: app/src/androidTest/java/me/varlez/movies/common/di/TestComponent.kt
package me.varlez.movies.common.di

import dagger.Component
import me.varlez.movies.common.di.component.MoviesComponent
import me.varlez.movies.movie.detail.MovieDetailsActivityBehaviorTest
import me.varlez.movies.movie.list.MovieListActivityBehaviorTest
import javax.inject.Singleton

/**
 * Dagger component for test purposes, using mock modules.
 */
@Singleton
@Component(modules = arrayOf(MockMoviesModule::class))
interface TestComponent : MoviesComponent {
    fun inject(movieListActivityBehaviorTest: MovieListActivityBehaviorTest)

    fun inject(movieDetailsActivityBehaviorTest: MovieDetailsActivityBehaviorTest)
}
